internal/controllers/admins.go

```diff
@@ -44,75 +45,47 @@ func (s Server) UpdateUsages(ctx echo.Context) error {
-			WHERE user_plans.effective_start_date <= CURRENT_DATE
-			AND user_plans.effective_end_date >= CURRENT_DATE
-			AND users.user_name = ?
-			AND resource_types.name = ?
-		`,
-		req.UserName,
-		req.ResourceType,
-	).Scan(&usageDetails).Error; err != nil {
+	err = s.GORMDB.Debug().Table("user_plans").Select("usages.*").Joins("JOIN usages ON user_plans.id=usages.user_plan_id").Joins("JOIN resource_types ON resource_types.id=usages.resource_type_id").Joins("JOIN quota ON user_plans.id=quota.user_plan_id").Joins("JOIN users ON users.id = user_plans.user_id").Where("resource_types.name=? AND users.user_name=?", req.ResourceType, req.UserName).Scan(&usageDetails).Error
+	if err != nil {
 		return ctx.JSON(http.StatusInternalServerError, model.ErrorResponse(err.Error(), http.StatusInternalServerError))
 	}
 	for _, usagerec := range usageDetails {
 		usagerec.UpdatedAt = effectivedate
 		value := req.UsageAdjustmentValue
-		if req.UpdateType == "sub" {
+		if req.UpdateType == "SET" {
 			value = -1 * value
+		} else if req.UpdateType == "ADD" {
+			value = 1 * value
 		}
 		usagerec.Usage += value
-		err := s.GORMDB.Debug().Updates(&usagerec).Error
+		err := s.GORMDB.Debug().Updates(&usagerec).Where("resource_type_id=?", resourceTypeID).Error
 		if err != nil {
 			return ctx.JSON(http.StatusInternalServerError, model.ErrorResponse(err.Error(), http.StatusInternalServerError))
 		}
 	}
-	return ctx.JSON(http.StatusOK, model.SuccessResponse("Success", http.StatusOK))
-}
-
-func (s Server) GetAllActiveQuotas(ctx echo.Context) error {
-	resource := ctx.QueryParam("resource")
-	resourcefilter := ""
-	if resource != "" {
-		resourcefilter = ` and resource_types.name = '` + resource + `'`
-	}
-	username := ctx.QueryParam("username")
-	usernamefilter := ""
-	if username != "" {
-		usernamefilter = ` and users.username = '` + username + `'`
+	var updateOperation = model.UpdateOperation{}
+	err = s.GORMDB.Debug().Where("name=?", req.UpdateType).Find(&updateOperation).Error
+	if err != nil {
+		return ctx.JSON(http.StatusInternalServerError, model.ErrorResponse(err.Error(), http.StatusInternalServerError))
 	}
-	plandata := []AdminQuotaDetails{}
-	err := s.GORMDB.Debug().Raw(`select users.user_name as user_name, users.id as user_id, plans.name as plan_name, quotas.quota, resource_types.name as resource_name, resource_types.unit from user_plans
-	join plans on plans.id = user_plans.plan_id	
-	join quotas on user_plans.id=quotas.user_plan_id
-	join resource_types on resource_types.id=quotas.resource_type_id
-	join users on users.id = user_plans.user_id
-	where
-	user_plans.effective_start_date <= cast(now() as date)and
-	user_plans.effective_end_date >=? ` + usernamefilter + resourcefilter).Scan(&plandata).Error
+	updateOperationID := *updateOperation.ID
+	var update = model.Update{ValueType: req.Unit, UpdatedBy: "Admin", Value: req.UsageAdjustmentValue, ResourceTypeID: &resourceTypeID, UpdateOperationID: &updateOperationID}
```

Comment:
It might be nice to reformat this so that the struct is a bit easier to read:

```suggestion
	var update = model.Update{
		ValueType: req.Unit,
		UpdatedBy: "Admin",
		Value: req.UsageAdjustmentValue,
		ResourceTypeID: &resourceTypeID,
		UpdateOperationID: &updateOperationID
	}
```

Running `gofmt` on the source will also cause it to automatically line up the field names and values, which is really nice.

As an aside, I wonder if there's a way to tell GORM to set a value from a `select` statement. It's just a thought, but being able to do everything in one round trip to the database might be nice as long as it's still possible to do all of the error checking that we need to do. If there's not a way to do that, this still works.

Reply:
Okay, I will try to find a way where select and set value.